src/main/java/server/Server.java

```diff
@@ -0,0 +1,102 @@
+package server;/*
+ * This Java source file was generated by the Gradle 'init' task.
+ */
+
+import Response.Response;
+
+import java.io.*;
+import java.net.*;
+import java.util.ArrayList;
+
+import static com.google.common.io.ByteStreams.toByteArray;
+
+public class Server {
+  private PrintWriter out;
+  private BufferedReader in;
+  private ServerSocket serverSocket;
+  private int PORT = 3000;
+  private Socket clientSocket;
+
+  public void go() throws IOException {
+    serverSocket = createSocket();
+
+    while (true) {
+      clientSocket = clientSocket(serverSocket);
+
+      out = createWriter(clientSocket);
+      in = createReader(clientSocket);
+      handleRequest(in, out);
+      closeConnection(in, out);
+      clientSocket.close();
+      System.out.println("Closing connection with testClient: " + clientSocket.getInetAddress());
+    }
+  }
+
+  public void handleRequest(BufferedReader in, PrintWriter out) {
+    try {
+      Response response = new Response();
+      String line;
+      ArrayList<String> responseText = new ArrayList();
+      while ((line = in.readLine()) != null) {
+        System.out.println(line);
+        responseText.add(line);
+          if (line.toLowerCase().contains("GET".toLowerCase())) {
+            out.write((response.res200("")));
+          } else if (line.toLowerCase().contains("POST".toLowerCase())) {
+            if (line.toLowerCase().contains("hello".toLowerCase())) {
+              out.write((response.res200("hello")));
+            } else if (line.toLowerCase().contains("POST".toLowerCase())) {
+              out.write((response.res200("goodbye")));
+            }
+            out.flush();
+        }
+      }
+    } catch (IOException ex) {
+      System.out.println(ex);
```

Comment:
This is a pretty common pattern - maybe there's an opportunity for a `ExitWithMessage` Java object?  So you can do something like `(new ExitWithMessage(ex, message)).perform()`?  Then if you do a different exit strategy later you can change it only once.

There's also a standard recommendation in Java land to _not_ rely on `try` / `catch` structures.  I don't personally mind them, but there are some good reasons to avoid them until absolutely necessary.  The `Exception` objects you get are generally useful in their own way.

One good thing to read here is [Exceptional Ruby](http://exceptionalruby.com/).  While it's a Ruby book, a lot of the good object practices from Ruby come from Java.